torch/lib/c10d/ProcessGroupNCCL.hpp

```diff
@@ -329,7 +328,8 @@ class ProcessGroupNCCL : public ProcessGroup {
               fut->markCompleted(at::IValue(cb()));
               // In case of chained then callback calls, thenFutCudaEvents
               // records callback's stream.
-              (*thenFutCudaEvents)[0].record(*futureNCCLCallbackStream_);
+              (*thenFutCudaEvents)[0].record(
+                  at::cuda::getDefaultCUDAStream(deviceIndex_));
```

Comment:
hmm, this should be current stream instead of default stream?

Reply:
Shall we also add a comment that when this cb is called, the current stream is `futureNCCLCallbackStream_`, so it's the same to record current stream here?

Reply:
> hmm, this should be current stream instead of default stream?

Yes, absolutely! I got confused, thanks! Luckily it seems the CI also picked that up (or at least I hope that was the reason of the failures, since there shouldn't be functional changes here).

> Shall we also add a comment...

I can, if you want, but this will change very shortly (first with the current stream becoming a stream from the ATen pool, and later this being possibly any stream at all).